### PR TITLE
Move to `dune` from `jbuilder`, prep 0.4.0 release

### PR DESCRIPTION
**Pull request by mor1**
Update to 4.06.1 involved a lot of mashing about with warnings etc.
Tests all pass though.

### CODE REVIEW

File: src/owl/linalg/owl_linalg_c.mli [hunk 1]
```diff
@@ -111,7 +111,7 @@ val bkfact : ?upper:bool -> ?symmetric:bool -> ?rook:bool -> mat -> mat * int32_
 
 val mpow : mat -> float -> mat
 
-val expm : mat -> mat
+val expm : mat -> mat [@@warning "-32"]
```
Comment (struktured) [inline src/owl/linalg/owl_linalg_c.mli]: You can just delete the duplicate definition instead. It's cleaner.

File: src/owl/dense/owl_dense_matrix_c.mli [hunk 2]
```diff
@@ -462,11 +462,11 @@ val mean : ?axis:int -> mat -> mat
 
 val mean': mat -> elt
 
-val var : ?axis:int -> mat -> mat
+val var : ?axis:int -> mat -> mat [@@warning "-32"]
```
Comment (struktured) [inline src/owl/dense/owl_dense_matrix_c.mli]: Just delete the duplicate definition of `var` instead.

File: src/owl/dense/owl_dense_matrix_c.mli [hunk 3]
```diff
@@ -48,11 +48,11 @@ val uniform : ?a:elt -> ?b:elt -> int -> int -> mat
...
 val gaussian : ?mu:elt -> ?sigma:elt -> int -> int -> mat
 
-val linspace : elt -> elt -> int -> mat
+val linspace : elt -> elt -> int -> mat [@@warning "-32"]
 
-val meshgrid : elt -> elt -> elt -> elt -> int -> int -> mat * mat
+val meshgrid : elt -> elt -> elt -> elt -> int -> int -> mat * mat [@@warning "-32"]
 
-val meshup : mat -> mat -> mat * mat
+val meshup : mat -> mat -> mat * mat [@@warning "-32"]
```
Comment (struktured) [inline src/owl/dense/owl_dense_matrix_c.mli]: Just delete the duplicate definition of `meshup`

**Comment by ryanrhymes** [comment]
Thank you @mor1 for the PR, we finally move to *dune* era :) also thank you @struktured for the comments.

**Pull request merged**